### DISCUSSION TITLE
How to fix wrong icon in material design extension?

### DISCUSSION THREAD

**Discussion by VahidEra** [MaterialDesignInXAML/MaterialDesignInXamlToolkit#3172] [Q&A] [answered]
Hi guys,
I am using material design in xaml toolkit in my wpf project.  And I also use material design extension, which is based on material design in xaml toolkit.
But there is a problem here, and this problem is that the icons in the material design extension are displayed incorrectly, like this:

![2023-04-23_171834](https://user-images.githubusercontent.com/27216710/233841141-73e2914e-5583-483b-9c57-e9919ff03711.png)

the icons should be like this:

![2023-04-23_171549](https://user-images.githubusercontent.com/27216710/233841247-e90f0121-3874-4bf1-8cae-0e5cf405dba2.png)

Of course, I must say that my project is based on .Net Framework 4.8, but the material design extension is compiled based on .NET Core 3.1 and .NET Framework 4.5.2.  Is this the problem and if so how do I fix it?

>**_Keep in mind that I changed the framework of my project to 4.5, but I ran into problems_**

**Comment by nicolaihenriksen** [comment, marked as answer]
@VahidEra This is a known/expected issue. See #3159 and #3125. The issue you're facing seems to be the exact problem of the latter. In short, there is a version mismatch.

**Reply by VahidEra** [reply]
Dear Nicolai, first thank you very much to helping me.
According to your guidance https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/issues/3125#issuecomment-1479786727, the only way to solve this problem is to use the Material Design version, according to which the Material Design Extension is compiled right?
In this case, how can I downgrade the material design of my project to a lower version?

**Reply by nicolaihenriksen** [reply]
@VahidEra well you could just use the Nuget Package Manager in Visual Studio to downgrade, or directly change it in your *.csproj manually.

However, beware that you will miss out on all new features and big fixes added after that version. So this is not ideal.

A better approach would be to get a new version of the Extension library which is compiled against a recent (preferably latest) version of MDIX. That, however, is up to the author of that library. This is an unfortunate side-effect when icons are directly referenced in code (not XAML). There are workarounds to this which I mentioned in the linked issue, but again that will require changes in the Extensions library.

**Reply by VahidEra** [reply]
@nicolaihenriksen Yes, I got it.  You are exactly right, because after changing the material design version, my project faced many problems.  So I prefer not to use material design extension at all.  Do you have a suggestion for a custom open directory?

**Reply by nicolaihenriksen** [reply]
@VahidEra Well, there are a few libraries out there to pick from I guess. Alternatively roll your own.

Another option is to fork the Material Design Extension library and build your own version of it?
I have a suspicion that you only need to change the [FileSysteminfoIconConverter.cs](https://github.com/spiegelp/MaterialDesignExtensions/blob/6e3e3d42707df61190ca30753a7f82e03b8a5c02/MaterialDesignExtensions/Converters/FileSystemInfoIconConverter.cs) class to load the PackIconKinds by reflection (as done in my comment [here](https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/issues/3159#issuecomment-1497981652)). I guess you could also consider creating a PR for the library with this change. It can/will have a small performance impact, but probably nothing noticeable for most use cases.

**Reply by VahidEra** [reply]
@nicolaihenriksen Dear Nicolai, thanks a million for your advice.  You are the best and the best never rest.
